code-challenges.test.js

```diff
@@ -18,54 +18,165 @@
 
+console.log(shuffleColors(colors1))
+console.log(shuffleColors(colors2))
 
 
 
-// b) Create the function that makes the test pass.
-
-
 
 
 
 // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with expect statements for each of the variables provided.
+//setting up test that will describe when cubedSum gets called 
+//it will take the argument and *** by itself and add each new number in that array to get a total. 
+describe("when sumUp is called", () =>{
+    it("if argument is cubeAndSum1, return 99",() =>{
+        const argument = cubeAndSum1;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(99)
+    })
+    it("if argument is cubeAndSum2, return 1125",() =>{
+        const argument = cubeAndSum2;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(1125)
+    })
+})
+
+
+// b) Create the function that makes the test pass.
+//setting up function that takes in an array of numbers as the argument
+
+
+const sumUp = (array) => {
+    let cubedArray =[]
+    for(let i= 0; i < array.length ; i++){
+        cubedArray.push(array[i] * array[i] * array[i])
+    }
+    return cubedArray.reduce((a,b) => (a + b))
+    
+}
+
+
 
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+console.log(sumUp(cubeAndSum1))
+console.log(sumUp(cubeAndSum2))
 
 
 
 
-// b) Create the function that makes the test pass.
-
 
 
 
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
+//setting up test that will describe when lowestAndBiggest gets called 
+//it will take in the arrgument of an array
+//return index 0 and the last index
+//for argument nums1 expect out to be [-8, 90]
+//for argument nums2 expect out to be [-59, 109]
+
+describe("when sortNums is called", () =>{
+    it("if argument is nums1, return -8, 90",() =>{
+        const argument = nums1;
+        const actualResults = sortNums(argument)
+        expect(actualResults).toEqual([-8, 90])
+    })
+    it("if argument is nums2, return -59, 109",() =>{
+        const argument = nums2;
+        const actualResults = sortNums(argument)
+        expect(actualResults).toEqual([-59, 109])
+    })
+})
+// b) Create the function that makes the test pass.
+//create a function that takes in an array and returns a mapped array with sorted array
+
+//filter the newSortedArray push index 0 and last index. 
+//return newSortedArray
+
+const sortNums = (array) => {
+
+    function sortNums(a,b) {
+        return a-b;}
+    
+      let sortedArray = array.sort(sortNums)
+      let firstAndLast = []
+      firstAndLast.push(sortedArray[0]);
+      firstAndLast.push(sortedArray[array.length-1])
+        return firstAndLast
```

Comment:
```suggestion
        return [sortedArray[0], sortedArray[array.length-1]]
```

You can initialize an array with values directly without having to do the work to push elements into a new array